components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragment.kt

```diff
@@ -70,6 +71,7 @@ internal class SitePermissionsDialogFragment : AppCompatDialogFragment() {
 
     internal var feature: SitePermissionsFeature? = null
     internal var userSelectionCheckBox: Boolean = false
+    internal var permissionRequest: PermissionRequest? = null
```

Comment:
I think it will be better if we don't keep this reference as mentioned on https://github.com/mozilla-mobile/android-components/pull/8555#discussion_r497764502 this value will be not available in situations where Android recreates the fragment, using a null able type we are getting away from the null pointer exception, but we are introducing a new bug:

1. With don't keep activities active, trigger the prompt.
2. Leave the app and go to another app.
3. Return to the app, you will see the prompt being recreated.
4. Tap either "Don't allow" or "Allow" buttons.

None of these actions are taking place, this creates a false expectancy for users to believe the permission was `allowed` or `denied` and later the dialog is going to show again.  

 The issue is that `permissionRequest` is `null`, because the fragment was recreated and this value wasn't re assigned to avoid that I believe we could use an alternative as suggested on https://github.com/mozilla-mobile/android-components/pull/8555#discussion_r497764502 of storing an identifier of the permission, in the fragment arguments, and later passing it to the `feature`, similar to what we are doing with `sessionId`,  as an example you could take a look at all the state of this fragment like `sessionId`, `icon`, `dialogGravity` and others, all of them are following the same pattern to avoid similar issues, you could find more info about this pattern [here](https://stackoverflow.com/a/9245510/5533820) and for a more expanded version [here](https://developer.android.com/reference/android/app/Fragment.html#lifecycle).

If you would like to schedule a meeting to chat about it just let me know happy to help :)